### ISSUE TITLE
Style/GuardClause suggest changing `if`/`elsif` to two `if` blocks 

### ISSUE THREAD

**Issue by ojab** [rubocop-hq/rubocop#5781]
Given code (conditions and body are shortened, in real life example it's not possible to use modifier `if`, i. e. `raise 'First' if 1 > 2` due to line length)
```ruby
def m
  if 1 > 2
    raise 'First'
  elsif 2 > 1
    raise 'Second'
  end
  puts 'passed'
end
```
## Expected behavior
no offences

## Actual behavior
```
Offenses:

/tmp/1.rb:2:3: C: Style/GuardClause: Use a guard clause instead of wrapping the code inside a conditional expression.
  if 1 > 2
  ^^

1 file inspected, 1 offense detected
```
if code will be changed to 
```ruby 
def m
  if 1 > 2
    raise 'first'
  end
  if 2 > 1
    raise 'second'
  end
  puts 'passed'
end
```
then `Style/GuardClause` doesn't complain, but I don't see how this code is better.
I assume that it's a bug and `if`/`elsif` case should not be counted as offence.

## RuboCop version
`0.54.0 (using Parser 2.5.0.5, running on ruby 2.5.0 x86_64-linux)`

**Comment by bquorning** [comment]
I see what you’re saying. The entire if/elsif block acts as one big guard clause.

That said, I think your example could be rewritten this way:

```rb
def m
  raise 'First' if 1 > 2
  raise 'Second' if 2 > 1

  puts 'passed'
end
```

What do you think?

**Comment by ojab** [comment]
As I've said, unfortunately resulting lines will be too long, so it's not an option.

**Comment by timon** [comment]
I have similar issue: `Style/GuardClause` code complaints for the following code:

```ruby
  def do_something(arg)
    if arg.predicate_that_should_be_false?
       raise ArgumentError, "You gave me something I don't want to handle"
    elsif calculate_this_way?
       calculate_this_way
    else
       calculate_another_way
    end

    self
end
```

raise + check statements are long enough to not fit into one line together; however no offence is registered if the guard condition is moved into separate `if` block and the rest of the `elsif` branches cases into another if:

```ruby
  def do_something(arg)
    if arg.predicate_that_should_be_false?
       raise ArgumentError, "You gave me something I don't want to handle"
    end

    if calculate_this_way?
       calculate_this_way
    else
       calculate_another_way
    end

    self
end
```

Should not GuardClause cop consider `elsif` acceptable when guard clause is too long to fit on one line?
